chronos/internal/asyncengine.nim

```diff
@@ -10,6 +10,94 @@
 
 {.push raises: [].}
 
+## This module implements the core asynchronous engine / dispatcher.
+##
+## The dispatcher acts as a kind of loop that wires events coming from the
+## operating system to tasks scheduled by the program. You must call
+## ``poll`` on it (or a function which does so for you such as ``waitFor`` or
+## ``runForever``) in order to poll for any outstanding events.
+##
+## The underlying implementation is based on epoll on Linux, IO Completion Ports
```

Comment:
`epoll` or `poll` on Linux, `kqueue` or `poll` on BSDs and MacOS and IOCP on Windows and do not works. No other OS supported.

Reply:
81a6584a44c69c03c47117c848dbbe63fac79fb5